### PR TITLE
[CDF-20663] Upgrade cognite-sdk with Writable Classes

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Please describe the change you have made.

## Checklist

- [x] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [x] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cdf_tk/load.py [hunk 1]
```diff
@@ -296,43 +316,33 @@ def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> Capability | lis
...
-    def get_id(cls, item: T_CogniteResource | T_CogniteResourceWrite) -> T_ID:
+    def get_id(cls, item: T_WriteClass | T_WritableCogniteResource) -> T_ID:
         raise NotImplementedError
 
-    @staticmethod
-    def fixup_resource(local: T_CogniteResourceWrite, remote: T_CogniteResource) -> T_CogniteResourceWrite:
-        """Takes the local (to be pushed) and remote (from CDF) resource and returns the
-        local resource with properties from the remote resource copied over to make
-        them equal if we should consider them equal (and skip writing to CDF)."""
-        return local
```
Comment (doctrino) [inline cognite_toolkit/cdf_tk/load.py:307]: This is no longer necessary as it implemented through the `cognite-sdk` with the `.as_write` method

File: tests/test_approval_modules_snapshots/cdf_apm_simple_data_model.yaml [hunk 2]
```diff
@@ -356,7 +362,6 @@ Transformation:
   ignoreNullFields: true
   isPublic: true
   name: asset:oid:workmate:apm_simple:load_asset2children
-  ownerIsCurrentUser: true
```
Comment (doctrino) [inline tests/test_approval_modules_snapshots/cdf_apm_simple_data_model.yaml:359]: This is a server set property, which was set by a default in the read class  which is no longer there.

**Review by gregertw — COMMENTED** [review]
See comments

File: cognite_toolkit/cdf_tk/load.py [hunk 3]
```diff
@@ -537,23 +541,32 @@ def create(self, items: GroupList, drop: bool, filepath: Path) -> GroupList:
             return GroupList([])
         # We MUST retrieve all the old groups BEFORE we add the new, if not the new will be deleted
         old_groups = self.client.iam.groups.list(all=True)
-        created = cast(GroupList, self.client.iam.groups.create(to_create))
+        old_group_by_names = {g.name: g for g in old_groups.as_write()}
+        changed = []
+        for item in to_create:
+            if (old := old_group_by_names.get(item.name)) and old == item:
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite_toolkit/cdf_tk/load.py:547]: yes

File: cognite_toolkit/cdf_tk/load.py [hunk 4]
```diff
@@ -855,9 +876,9 @@ def delete(self, ids: SequenceNotStr[str], drop_data: bool) -> int:
         except CogniteNotFoundError as e:
             return len(ids) - len(e.not_found)
 
-    def create(self, items: TransformationScheduleList, drop: bool, filepath: Path) -> TransformationScheduleList:
+    def create(self, items: TransformationScheduleWriteList, drop: bool, filepath: Path) -> TransformationScheduleList:
         try:
-            return cast(TransformationScheduleList, self.client.transformations.schedules.create(items))
+            return self.client.transformations.schedules.create(list(items))
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite_toolkit/cdf_tk/load.py:881]: Yes, but lets leave that for another PR.

**Review by gregertw — APPROVED** [review]
Alright, good to go! :-D

**Pull request merged**